src/backend/PortabilityService.ConfigurationService/Controllers/ConfigurationController.cs

```diff
@@ -0,0 +1,115 @@
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Localization;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using System;
+using System.Linq;
+
+namespace PortabilityService.ConfigurationService.Controllers
+{
+    [Route("api/[controller]")]
+    public class ConfigurationController : Controller
+    {
+        private readonly IConfiguration _configuration;
+        private readonly IHostingEnvironment _hostingEnvironment;
+        private readonly IStringLocalizer<ConfigurationController> _localizer;
+        private readonly ILogger _logger;
+        private readonly string _baseConfigSectionName;
+
+        public ConfigurationController(IConfiguration configuration,
+                                       IHostingEnvironment hostingEnvironment,
+                                       IStringLocalizer<ConfigurationController> localizer,
+                                       ILogger<ConfigurationController> logger = null,
+                                       string baseConfigSectionName = "PortabilityServiceSettings")
+        {
+            _configuration = configuration ?? throw new ArgumentNullException(nameof(configuration));
+            _hostingEnvironment = hostingEnvironment ?? throw new ArgumentNullException(nameof(hostingEnvironment));
+            _localizer = localizer ?? throw new ArgumentNullException(nameof(localizer));
+            _logger = logger ?? new NullLogger<ConfigurationController>();
+
+            if (string.IsNullOrEmpty(baseConfigSectionName))
+            {
+                throw new ArgumentNullException(nameof(baseConfigSectionName));
+            }
+            else
+            {
+                _baseConfigSectionName = baseConfigSectionName;
+            }
+        }
+
+        [HttpGet]
+        public ObjectResult Get()
+        {
+            _logger.LogInformation(BuildLogMessage(nameof(Get), _localizer[nameof(ConfigurationControllerResources.ReturningAllConfigSettings)].Value));
+            return Ok(_configuration.GetSection(_baseConfigSectionName).AsEnumerable());
+        }
+
+        [HttpGet("section/{sectionName}")]
+        public ObjectResult GetSection(string sectionName)
+        {
+            if (string.IsNullOrEmpty(sectionName))
+            {
+                var errorMessage = _localizer[nameof(ConfigurationControllerResources.SectionNameNullOrEmpty)].Value;
+                _logger.LogError(BuildLogMessage(nameof(GetSection), errorMessage));
+                return BadRequest(errorMessage);
+            }
+
+            _logger.LogInformation(BuildLogMessage(nameof(GetSection), _localizer[nameof(ConfigurationControllerResources.ReturningSectionSettings), sectionName].Value));
+            return Ok(_configuration.GetSection(sectionName));
```

Comment:
Should we get the section from within the base config section?

Reply:
Good catch. I updated both the GetSection and GetSectionSettingsList to return BadRequest if the sectionName does not start with the base